ibis/backends/bigquery/__init__.py

```diff
@@ -171,14 +170,16 @@ def _session_dataset(self):
     def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
         raw_name = op.name
 
-        project = self._session_dataset.project
-        dataset = self._session_dataset.dataset_id
+        session_dataset = self._session_dataset
+        project = session_dataset.project
+        dataset = session_dataset.dataset_id
 
-        if raw_name not in self.list_tables(database=(project, dataset)):
+        try:
+            self.client.get_table(bq.TableReference(session_dataset, raw_name))
```

Comment:
I think this fixes #9216.

Reply:
I'll move this change to another PR so it's not mixed up in this one.

Reply:
Moved to #9425.